core/core_kernel.cc

```diff
@@ -29,3 +30,78 @@ long long execute_kernel_busy_wait(const Kernel &kernel)
   }
   return acc;
 }
+
+// -- add by Yuankun
+void execute_kernel_memory(const Kernel &kernel)
+{
+    //currently use two src input
+    long long jump = kernel.jump;
+    long long N = kernel.scratch_bytes_per_task / 3;
```

Comment:
Does this need to be `/ sizeof(double)`? I.e. is N supposed to be number of bytes or number of doubles?